### PR TITLE
LPS-155704 Create a sample for FDS direct usage as a React component

### PR DESCRIPTION
**Pull request by markocikos**
### References

- [LPS-155704 Create a sample for FDS direct usage as a React component](https://issues.liferay.com/browse/LPS-155704)
- Related to #2325

### What is the goal of this PR?

We are adding a sample of FDS direct usage as a React component. All current usages in DXP are usages of `<frontend-data-set>` tags, but there is ongoing development to use it directly.

The main complication with this approach is that the setting if views, filters, actions and data providers needs to be done manually. In tags, it is done in tag classes, [under the hood](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-taglib/src/main/java/com/liferay/frontend/data/set/taglib/servlet/taglib/HeadlessDisplayTag.java#L415-L418). You only need to know to set [component properties that match FDS ID](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/java/com/liferay/frontend/data/set/sample/web/internal/view/CustomizedCardsFDSView.java#L28). Here, we need to set the `views` prop. In the sample, I did a manual invoke of serialization. This in not required, you can use completely frontend definition of view objects, a JS object passed as a prop. This was just the simplest thing to do for the sample.

I found one issue with this kind of a usage, fixed in a separate commit. There was an inconsistent handling of the case when there  are no actions. In the tags, we are setting [default empty array](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-taglib/src/main/java/com/liferay/frontend/data/set/taglib/servlet/taglib/HeadlessDisplayTag.java#L438-L439). In the direct usage, we are setting [`null`](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js#L846). When checking, empty array is not falsy and null is. We are now consistently setting an empty array as default.

There are likely more such inconsistencies, one of them might have caused #2325

/cc @rebsilva @miltonmc

### What does it look like?

screen

### Steps to reproduce

Deploy `frontend-data-set-sample-web`, add as a widget to any page.

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/SampleReactFrontendDataSet.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {FrontendDataSet} from '@liferay/frontend-data-set-web';
```
Comment (diegonvs) [inline modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/SampleReactFrontendDataSet.js:15]: It's so lovely to see FDS exported as a React component <3

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/Table.js [hunk 2]
```diff
@@ -210,9 +210,7 @@ function Table({dataLoading, items, itemsActions, schema, style}) {
 														item.actions) && (
 														<Actions
 															actions={
-																itemsActions ||
-																item.actions ||
-																item.actionDropdownItems
+																itemsActions
```
Comment (diegonvs) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/Table.js:213]: item.actions, item.actionDropdownItems and itemsActions are the same?

Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/Table.js:213]: No, but this condition makes no sense. `itemsActions` will never be falsy, as the default is an empty array:
```js
[] || 'foo' || 'bar' // []
```
We are not supporting action items definition in items themselves anyways. This is likely leftover code from some previous solution.

**Review by beltranrengifo — APPROVED** [review]
Hi @markocikos !
Nice to see this example in the sample module.
Glad you spotted the issue with the actions array.
It gave me no problems, everything seems fine.

I'm worrying a bit about [this](https://github.com/liferay-frontend/liferay-portal/pull/2325)
Let me know if I can help 🙌

**Comment by markocikos** [comment]
ci:test:relevant

**Comment by markocikos** [comment]
ci:forward

**Comment by markocikos** [comment]
ci:forward

**Comment by markocikos** [comment]
ci:stop

**Comment by markocikos** [comment]
ci:stop

**Comment by markocikos** [comment]
ci:test:relevant

**Comment by diegonvs** [comment]
ci:forward

**Pull request closed without merging**